airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -325,7 +326,7 @@ private static AirbyteStream removeIncrementalWithoutPk(final AirbyteStream stre
 
   @Override
   protected boolean isNotInternalSchema(JsonNode jsonNode, Set<String> internalSchemas) {
-    return false;
+    return !internalSchemas.contains(jsonNode.get(INTERNAL_SCHEMA_NAME).asText());
```

Comment:
We cannot make this changes, because we create another issue when user without any permission could see all tables. It happens because isNotInternalSchema() method is involved only if set of permitted tables and schemas is empty. In such case either user don't have permission or issue with fetching permitted tables, but not with that overridden method. Can you share list of privileges for used user.

Reply:
My bad, I'm not a java dev so I simply resorted to reverting the change made in order to make it work.

As reported in the previous issues, this is affecting many users and happens if _any_ table is not accessible to the user, so **all** of the tables seem to end up as not accessible. 
With this changes, it makes so only the correct tables are being accessed, and during my tests I didn't notice any unwanted access to other protected tables.

I don't have the proper privileges at hand, but if it's really necessary I could go after those.

Reply:
Also, how would an user without permission be able to see all tables? Wouldn't that be blocked by the database itself without any intervention from the application's code?